src/actions/rooms.js

```diff
@@ -31,9 +31,14 @@ export const joinRoomsChannel = (onSuccess, onError) => (dispatch, getState) =>
     channel.on('rooms:public:created', (data) => {
       dispatch(addPublicRoom(data))
     })
+
     channel.on('starred_messages', (data) => {
-      dispatch(replaceStarMessages(data['starred_messages']))
+      const cased = camelize(data['starred_messages'])
+      const asObject = listToObject(cased, 'messageId')
+
+      dispatch(replaceStarMessages(asObject))
```

Comment:
Moved the data transforming into the Action Creator instead of the reducer. It's preferable to do the transform as close to the receiving the data as possible.